src/containers/Application/ApplicationPageWrapper.tsx

```diff
@@ -44,10 +57,42 @@ const ApplicationPageWrapper: React.FC = () => {
         push,
         replace,
       })
+
+      if (sectionCode && page)
+        setCurrentSection(templateSections.find(({ code }) => code === sectionCode))
+    }
+  }, [isApplicationLoaded, sectionCode, page])
+
+  useEffect(() => {
+    const progressStructure = buildProgressInApplication({
+      elementsState,
+      responses: responsesFullByCode,
+      templateSections,
+      isLinear: application?.isLinear as boolean,
+      currentSection: currentSection?.index as number,
+      currentPage: Number(page),
+    })
+    setProgressInApplication(progressStructure)
+    setProcessingValidation(false)
+  }, [elementsState, currentSection, page])
+
+  const validateCurrentPage = (): boolean => {
+    if (!application || !currentSection || !page) {
+      console.log('Problem to validate - Undefined parameters')
+      return false
     }
-  }, [application])
+    setProcessingValidation(true)
+    const validation = validatePage({
+      elementsState: elementsState as ApplicationElementStates,
+      responses: responsesFullByCode as ResponsesFullByCode,
+      sectionIndex: currentSection?.index as number,
+      page: Number(page),
+      checkEmpty: true,
```

Comment:
what is the use case of checkEmpty parameter, it's not apparent from reading the code

Reply:
The `checkEmpty` is used for the **strict** validation. The default then is **loose** validation, which means that the validatePage is not checking for empty responses. I'll move this to a helper file and add some comments to this.

Reply:
Oh, sorry, it's already in a helper file. I'll just add a comment explaining what I just did here. Then we can improve this function as needed.